pkg/oci/oci.go

```diff
@@ -553,18 +560,26 @@ func getMetadataFromManifest(ctx context.Context, target oras.Target, manifest *
 	return metadata, nil
 }
 
-func getEbpfProgramFromManifest(ctx context.Context, target oras.Target, manifest *ocispec.Manifest) ([]byte, error) {
-	if len(manifest.Layers) != 1 {
-		return nil, fmt.Errorf("expected exactly one layer, got %d", len(manifest.Layers))
+func getLayerFromManifest(ctx context.Context, target oras.Target, manifest *ocispec.Manifest, mediaType string) ([]byte, error) {
+	var layer ocispec.Descriptor
+	layerCount := 0
+	for _, l := range manifest.Layers {
+		if l.MediaType == mediaType {
+			layer = l
+			layerCount++
+		}
+	}
+	if layerCount != 1 {
+		return nil, fmt.Errorf("expected exactly one layer with media type %q, got %d", mediaType, layerCount)
```

Comment:
This makes the `wasm` layer mandatory and that shouldn't be the case

Reply:
Thanks. I fixed it and also updated the reference documentation accordingly.